### PR TITLE
Fix input prediction.

### PR DESCRIPTION
**Pull request by kieve**
This was reading from what used to be an object with booleans, to an object with a bitfield. It was not updated correctly.

### CODE REVIEW

File: client/js/player/controls.js [hunk 1]
```diff
@@ -113,6 +113,21 @@ return function Controls(elm) {
...
+		},
+		right: function() {
+			return (this.controlFlags & ActionMasks.right) > 0;
+		},
+		jump: function() {
+			return (this.controlFlags & ActionMasks.jump) > 0;
+		},
 		activateLeft: function() {
 			return (this.controlFlags & ActionMasks.activateLeft) > 0;
 		},
```
Comment (crazy2be) [inline client/js/player/controls.js]: Might as well make this generic:
```	
    for (var action in ActionMasks) {
		actions[action] = function(action) {
			return function () {
				return (this.controlFlags & ActionMasks[action]) > 0;
			}
		}(action);
	};
````

File: server/lib/game/player.go [hunk 2]
```diff
@@ -292,8 +292,6 @@ func (p *Player) simulateMovement(dt float64, controls ControlState) {
 		body.Vel.Y = 0
 		if controls.Jump() {
 			body.Vel.Y = 6
-		} else {
-			body.Vel.Y = 0
```
Comment (crazy2be) [inline server/lib/game/player.go]: Eh?

Comment (kieve) [inline server/lib/game/player.go]: Duplicated. Look at the line right above the if.

File: client/js/entities/model/bioticState.js [hunk 3]
```diff
@@ -28,9 +28,7 @@ return function BioticState(entityState, health) {
 
 	self.prettyCloseTo = function (other) {
 		return closeVec(self.entityState.body.pos, other.entityState.body.pos)
-			&& closeVec(self.entityState.body.dir, other.entityState.body.dir)
-			&& closeVec(self.entityState.body.vel, other.entityState.body.vel)
-			&& close(self.health.life, other.health.life);
+			&& closeVec(self.entityState.body.vel, other.entityState.body.vel);
```
Comment (crazy2be) [inline client/js/entities/model/bioticState.js]: This doesn't seem correct. Our directions in the entityState should match, right?

File: server/lib/game/player.go (hunk 2 above)
Comment (crazy2be) [inline server/lib/game/player.go]: Oh, I see.

File: client/js/entities/model/bioticState.js (hunk 3 above)
Comment (kieve) [inline client/js/entities/model/bioticState.js]: They never will. Ever. Either the direction isn't changing (and thus, they match) or the client is changing the direction, and then it won't ever match.

We would want this in a case where the server forces the client to look a particular direction... but we don't have anything like that right now.

Comment (crazy2be) [inline client/js/entities/model/bioticState.js]: Like these are not current directions but past directions. The mapping (controls, position) -> (position, velocity, direction) should be the same on both client and server, right?

**Comment by crazy2be** [comment]
@yeerkkiller1 will save us with TypeScript!!!!!!!!!!!!

:ship: :it:

**Pull request merged**